Updated Swagger as per latest changes.

amoghnatu opened the pull request:
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [X] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [X] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [X] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

Commit:
Updated Swagger as per latest changes.

amoghnatu committed:
Added description field

Added the missing 'description' field in CreatedBy element

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

amoghnatu commented:
**WaitForARMFeedback**
Unable to add the label "WaitForARMFeedback" so addint it as a comment

yungezz commented on specification/databricks/resource-manager/Microsoft.Databricks/stable/2018-04-01/databricks.json:
```diff
@@ -391,16 +391,20 @@
           }
+        },
+        "createdBy": {
+          "description": "Indicates the Object ID, PUID and Application ID of entity that created the workspace.",
```
is this readonly property?

amoghnatu replied:
Yes they are..

yungezz commented on specification/databricks/resource-manager/Microsoft.Databricks/stable/2018-04-01/databricks.json:
```diff
@@ -391,16 +391,20 @@
           "$ref": "#/definitions/ProvisioningState",
           "description": "The workspace provisioning state."
         },
-        "uiDefinitionUri": {
```
remove property from stable version is breaking change, suggest to do it in new api version. Just to confirm, is this a Swagger file bug fixing, fixing Swagger file to align with existing service behavior? or new change?

amoghnatu replied:
this is fixing swagger file to align with existing service behavior.

amoghnatu replied:
@yungezz After discussing with the team internally, I have reverted this change.

ravrams commented on specification/databricks/resource-manager/Microsoft.Databricks/stable/2018-04-01/examples/WorkspaceCreate.json:
```diff
@@ -35,15 +35,32 @@
-          "provisioningState": "Created",
-          "uiDefinitionUri": "https://path/to/workspaceCreateUiDefinition.json"
+          "provisioningState": "Accepted",
+          "authorizations": [
```
Are these response for create request or create response

amoghnatu replied:
this is the response for Create request

yungezz commented on specification/databricks/resource-manager/Microsoft.Databricks/stable/2018-04-01/databricks.json:
```diff
@@ -615,6 +619,25 @@
       "description": "Custom Parameters used for Cluster Creation."
     },
+    "CreatedBy": {
+      "properties": {
```
Then pls add readonly true property. thanks.

amoghnatu replied:
Added.

amoghnatu committed:
added readonly attribute

added readonly attribute to properties

AutorestCI commented:
/azp run automation - sdk

ravrams commented:
Can you please update with below changes as discussed offline.
1. New property 'properties.parameters.enableFedRampCertified' 
2. Returning 'properties.CreatedDateTime'
3. VNet peering in the operation list
4. AML workspace id in the example should be a resource uri
5. Sku name value should be allowed value (in the example)
6. Revert changes to remove properties
7. Update examples as well.

amoghnatu committed:
Fixed review comments

Fixed all the below review comments.
1. New property 'properties.parameters.enableFedRampCertified'
2. Returning 'properties.CreatedDateTime'
3. VNet peering in the operation list
4. AML workspace id in the example should be a resource uri
5. Sku name value should be allowed value (in the example)
6. Revert changes to remove properties
7. Update examples as well.

AutorestCI commented:
/azp run automation - sdk

amoghnatu commented:
@yungezz can you confirm when we can merge this PR?

amoghnatu committed:
Fixed model validation errors

Fixed model validation errors.

AutorestCI commented:
/azp run automation - sdk

amoghnatu committed:
fixed typo

AutorestCI commented:
/azp run automation - sdk

amoghnatu committed:
updated datetime format

AutorestCI commented:
/azp run automation - sdk

amoghnatu committed:
fixed error response definition path

AutorestCI commented:
/azp run automation - sdk

yungezz commented:
LGTM. waiting for ARM feedback.

amoghnatu committed:
fixed prettier issues

AutorestCI commented:
/azp run automation - sdk

ravbhatnagar commented:
adding new writable properties to the schema requires a new api-version. But looks like these changes have already been made on service side. Signing off.

amoghnatu committed:
updated readonly on the properties

AutorestCI commented:
/azp run automation - sdk

ravrams commented on specification/databricks/resource-manager/Microsoft.Databricks/stable/2018-04-01/examples/WorkspaceGetParameters.json:
```diff
@@ -67,13 +70,42 @@
+            },
+            "enableFedRampCertified": {
+              "type": "Bool",
+              "value": true
```
Value should be false by default

amoghnatu committed:
Changed enableFedRamp value

AutorestCI commented:
/azp run automation - sdk

ravrams commented on specification/databricks/resource-manager/Microsoft.Databricks/stable/2018-04-01/examples/WorkspaceCreateWithParameters.json:
```diff
@@ -74,10 +79,31 @@
               "type": "String",
               "value": "10.139"
+            },
+            "enableFedRampCertified": {
```
Property name is enableFedRampCertification

amoghnatu committed:
Renamed property

AutorestCI commented:
/azp run automation - sdk

amoghnatu committed:
removed unnecesary 404 code

AutorestCI commented:
/azp run automation - sdk

amoghnatu committed:
removed 404 response in example

AutorestCI commented:
/azp run automation - sdk

azuresdkci commented:
Can one of the admins verify this patch?

amoghnatu committed:
removed 404

AutorestCI commented:
/azp run automation - sdk

amoghnatu committed:
updated autorest

AutorestCI commented:
/azp run automation - sdk

yungezz commented:
hi @amoghnatu pls ping here when the PR is ready to merge, thanks.

amoghnatu committed:
merged latest master

AutorestCI commented:
/azp run automation - sdk

amoghnatu committed:
updated package version

AutorestCI commented:
/azp run automation - sdk

akning-ms commented:
@amoghnatu, can you restore your change for package.json and package-lock.json? it will impact our pipeline which is not allow to change.

amoghnatu committed:
Reverted package.json and package lock json changes

AutorestCI commented:
/azp run automation - sdk

amoghnatu committed:
added workspaceId and workspaceUrl properties to existing API version.

AutorestCI commented:
/azp run automation - sdk

yungezz commented on specification/databricks/resource-manager/Microsoft.Databricks/stable/2018-04-01/databricks.json:
```diff
@@ -410,6 +410,14 @@
           "description": "Specifies the date and time when the workspace is created.",
           "$ref": "#/definitions/CreatedDateTime"
+        },
+        "workspaceId": {
```
adding properties to existing version is breaking change. need to do it in new version

amoghnatu replied:
@yungezz We're going to backfill all existing databricks workspaces with these 2 properties. Any new workspaces created with this api-version will get these 2 properties by default. And all existing workspaces, we're going to backfill them next week. So this shouldn't technically remain as a breaking change after the backfilling is done. Also, since the backfill process is scheduled for next week, we request you to put a "DoNotMerge" label until we complete the process.

yungezz replied:
HI @amoghnatu, thanks for sharing this. Adding new properties in response will break the GET-PUT pipeline, here's guideline of breaking change https://github.com/Azure/adx-documentation-pr/wiki/Breaking-changes-guidelines#new-property-added-to-response

amoghnatu replied:
@yungezz Thanks for mentioning this. However, in our case, these 2 new properties are **readonly** and the customer will not be able to set them. Also, these are not available in the template so user cannot set them to provide as input. I have updated the spec to include readonly=true for these 2 properties. Kindly check.

yungezz commented:
since there're new properties added after ARM signoff. need ARM review again. Wait for ARM review.

amoghnatu committed:
Added readonly to new properties.

AutorestCI commented:
/azp run automation - sdk

akning-ms commented:
Hi @amoghnatu, Looks there is some change for package-lock.json file in your PR which is not supposed in swagger PR. can you remove this file change in this PR? Thanks

amoghnatu committed:
Reverted change in package.lock.json

AutorestCI commented:
/azp run automation - sdk

amoghnatu commented:
@akning-ms Thanks for pointing it out. I have removed the file.

amoghnatu committed:
Added missing 200 response on delete.

AutorestCI commented:
/azp run automation - sdk

amoghnatu committed:
fixed model validation error.

AutorestCI commented:
/azp run automation - sdk

ravbhatnagar commented:
signing off from ARM side.

yungezz commented:
hi @amoghnatu, could you pls fix CI failure, then the PR ok to merge.

amoghnatu committed:
Removed FedRAMP parameter from api-version

AutorestCI commented:
/azp run automation - sdk

yungezz commented:
hi @amoghnatu could you pls fix CI?

yungezz commented:
ping @amoghnatu

AutorestCI commented:
/azp run automation - sdk
